lib/src/provider/network_vector_tile_provider.dart

```diff
@@ -26,19 +28,19 @@ class NetworkVectorTileProvider extends VectorTileProvider {
   ///  confused with the maximum zoom of the map widget. The map widget will
   ///  automatically use vector tiles from lower zoom levels once the maximum
   ///  supported by this provider is reached.
-  NetworkVectorTileProvider(
-      {required String urlTemplate,
-      this.type = TileProviderType.vector,
-      this.httpHeaders,
-      this.tileOffset = TileOffset.DEFAULT,
-      this.maximumZoom = 16,
-      this.minimumZoom = 1})
-      : _urlProvider = _UrlProvider(urlTemplate);
+  NetworkVectorTileProvider({
```

Comment:
Thanks for the improvement. Would we be better off wrapping `NetworkVectorTileProvider` instead of making it extensible? e.g.

```
class SomeSpecialVectorTileProvider extends VectorTileProvider {
  NetworkVectorTileProvider _delegate;

  ...
}
```

Reply:
I still wouldn't be able to access the URL template (as far as I understand), so I'd need to create my own equivalent `_UrlProvider` or at least define a method as I have done here anyway.

The bigger issue is actually wrapping it. The style reader returns your class, not the wrapper.

Reply:
can you help me to understand the use-case? What is FMTC?

Reply:
https://github.com/greensopinion/flutter-vector-map-tiles/pull/231#issuecomment-2614575671 just posted :D